cassandra-mesos-scheduler/src/test/java/io/mesosphere/mesos/frameworks/cassandra/scheduler/CassandraSchedulerTest.java

```diff
@@ -661,4 +772,160 @@ public void testResourceList_doesNotIncludesDiskWhenBelow0() throws Exception {
+                .addResources(Protos.Resource.newBuilder()
+                        .setName("disk")
+                        .setRole("*")
+                        .setType(Protos.Value.Type.SCALAR)
+                        .setScalar(Protos.Value.Scalar.newBuilder().setValue(8192)))
+                .build();
+
+        final List<Protos.Resource> resources = CassandraScheduler.resourceList(
+                TaskResources.newBuilder()
+                        .setCpuCores(0.5)
+                        .setMemMb(512)
+                        .setDiskMb(1000)
+                        .build(),
+                "someRole",
+                offer);
+
+        assertThat(resources).hasSize(3);
+        assertThat(resources).contains(ProtoUtils.cpu(0.5, "*"));
+        assertThat(resources).contains(ProtoUtils.mem(512, "*"));
+        assertThat(resources).contains(ProtoUtils.disk(1000, "*"));
+    }
+
+    @Test
+    public void testPorts_doesReturnPortsWithTheRoleFromOffers() {
+        Protos.Offer offer = Protos.Offer.newBuilder()
+                .setFrameworkId(frameworkId)
+                .setHostname("somehost.name")
+                .setId(Protos.OfferID.newBuilder().setValue(randomID()))
+                .setSlaveId(Protos.SlaveID.newBuilder().setValue("someslave").build())
+                .addResources(Protos.Resource.newBuilder()
+                        .setName("cpus")
+                        .setRole("*")
+                        .setType(Protos.Value.Type.SCALAR)
+                        .setScalar(Protos.Value.Scalar.newBuilder().setValue(8d)))
+                .addResources(Protos.Resource.newBuilder()
+                        .setName("mem")
+                        .setRole("*")
+                        .setType(Protos.Value.Type.SCALAR)
+                        .setScalar(Protos.Value.Scalar.newBuilder().setValue(8192)))
+                .addResources(Protos.Resource.newBuilder()
+                        .setName("disk")
+                        .setRole("*")
+                        .setType(Protos.Value.Type.SCALAR)
+                        .setScalar(Protos.Value.Scalar.newBuilder().setValue(8192)))
+                .addResources(Protos.Resource.newBuilder()
+                        .setName("ports")
+                        .setRole("*")
+                        .setType(Protos.Value.Type.RANGES)
+                        .setRanges(Protos.Value.Ranges.newBuilder()
+                                .addRange(Protos.Value.Range.newBuilder().setBegin(7000).setEnd(10000))))
+                .build();
+
+        ImmutableList<Long> ports = of(8080l, 9090l);
+        final Protos.Resource resources = CassandraScheduler.ports(ports, "someRole", offer);
+
+        assertThat(resources.getName()).isEqualTo("ports");
+        assertThat(resources.getRanges().getRangeList()).contains(range(8080),range(9090));
+        assertThat(resources.getRole()).isEqualTo("*");
+    }
+
+    private Protos.Value.Range range(int value) {
```

Comment:
Please move down with other helper methods so that test methods are kept together.